### PR TITLE
Lab Cloud vs. on-premise

### PR DESCRIPTION
**Pull request by darienmt**

### CODE REVIEW

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Lab : Cloud vs. on-premise
+
+* Let us list a few activities we do on a [Software Development Life Cycle](https://en.wikipedia.org/wiki/Systems_development_life_cycle):
+
+  * Development platform (Ex. Java, .Net, Node.js, Python, etc.)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: Ex. => e.g.:

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  * Storage:
+    * Deployment articraft (Ex. .dlls, .war, .ear, files, etc)
+    * Data Storage (Ex. RDBMS, NoSQL Database, files)
+    * Source control (Ex. Git, SVN, Team Foundation Server)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: Brackets are everywhere:
Source control (Ex. Git, SVN, Team Foundation Server) => 
Source control, e.g.: Git, SVN, Team Foundation Server, etc

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  * Non-functional requirements:
+    * Traffic estimation (Ex. request/second, jobs/minute)
+    * Response time (Ex. max response time [ms])
+    * Supported number of users
+    * Up-time requirements (Ex. Can we have maintenance windows? How many nines we need?)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: How many nines we need? =>
How many nines do we need?

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    * Disaster recovery plans (Ex. We need to store data for a year)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: Disaster recovery vs. Data Retention, incorrect example, breakdown to two points:
Disaster Recovery: how fast can we afford to stay offline?
Data Retention: how long should we keep our data back in time?

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    * Data locality (Ex. We need to be available worldwide but data is consumed locally, or a user in London should be able to access data from New York)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: Data locality, e.g.: we need to make the system available worldwide, but data is consumed locally. Should a user in London, UK be able to access data from New York, NY?

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  * Deployment
+    * Deployment pipeline (Ex. continuous deployment, canary deployment, blue-green deployment)
+    * QA, Staging and Production environments.
+  * Network requirements
+    * Load balancers (Ex. F5)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: F5? Maybe Cisco is a more appropriate example?

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    * Firewall to create a [DMZ](https://en.wikipedia.org/wiki/DMZ_(computing) )
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: Links should be opening in a new window:
[DMZ](https://en.wikipedia.org/wiki/DMZ_(computing) =>
```
<a href="https://en.wikipedia.org/wiki/DMZ_(computing)" target="_blank">DMZ</a>
```

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    * DNS setup
+    * Security certificates
+
+
+* Taking all of this points into account (and any others you might consider) put a time estimation on how much time will take to provision the infrastructure.
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: Taking all the points into an account and any others you might consider, put a timeframe estimate on how much time will take to provision the needed infrastructure

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    * Load balancers (Ex. F5)
+    * Firewall to create a [DMZ](https://en.wikipedia.org/wiki/DMZ_(computing) )
+    * DNS setup
+    * Security certificates
+
+
+* Taking all of this points into account (and any others you might consider) put a time estimation on how much time will take to provision the infrastructure. 
+
+
+* OK, everything is running what happens if:
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: OK, everything is running what happens if =>
OK, everything is running for now. What happens when

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  * The project failed
+  * The load is **much more** than anticipated
+  * The load is **much less** than anticipated
+  * This new architect want to change part of technology stack
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: A new architect wants to change part of the technology stack for very compelling reasons

File: content/chapter-01/LabCloudVsOnPremise.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  * The Executive Vice-President wish to deploy to the Australia division(and you are not in Australia)
```
Comment (vkhazin) [inline content/chapter-01/LabCloudVsOnPremise.md]: An Executive Vice-President wishes to deploy the system to Australia division and you are not in Australia

**Review by vkhazin — CHANGES_REQUESTED** [review]
Added comments for review

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 1 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 2 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 3 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 4 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 5 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 6 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Added Cisco and Citrix

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 7 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: It is opening a new page with markdown syntax, e.g.: [Something](http://something)

Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 8 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

File: content/chapter-01/LabCloudVsOnPremise.md (hunk 9 above)
Comment (darienmt) [reply content/chapter-01/LabCloudVsOnPremise.md]: Done

**Pull request merged**